Add Chris API

Christley opened the pull request:
Exposes a few data points to be used with website building, tournament information or just nerd information. Spiritual successor to Morg HTTP Client with some added functionality.

Shows some information about your stats, quests, inventory, equipment and some miscellaneous events like player weight and combat level

YvesW commented:
Please contribute to the original plugin instead of submitting a new plugin. That way there is less pluginhub clutter and the current users of the plugin can benefit from potential improvements and bugfixes.
Additionally, a (large) part of your code seems to be copied from Morg Http Client, but I do not see any attribution to the author.

Furthermore, Morg Http Client has been unavailable for 1 day (see the issue you created), and someone submitted a PR to the repo that likely fixes the plugin?

Christley commented:
I have completely rewritten the plugin now. Not a single line of code has been taken from any other plugin.

I've added four new endpoints being quests, combat, account info, bank.

I've added multithreading to the endpoints so if you call them fast enough it shouldn't crash the entire plugin.

I've extended the equipment and inventory endpoints with item names and slot name in the equipment handler.

I've extended the events endpoint to show last 5 chat messages and special attack energy and if you're idle or not.

I've extended the skills endpoint to show the amount of boost your skills have right now.


This is not even close to the same plugin as morg anymore and uses no code from that plugin either. If this can't be accepted because it shows some of the data that morg did, then it's weird that other plugins have been accepted which also show similar data

cdfisher commented:
To reiterate what was said earlier,

> We'd rather you try and contribute to an existing plugin.

The preference would be for you to contribute to the similar plugin that already exists. If it remains abandoned, transfer of ownership to a new maintainer who would be active would generally be the next step.

Christley commented:
Alright since it's obvious that this won't ever be implemented, how long do I need to wait before either me or someone else is given ownership of morg?

Currently his plugin contains some datapoints that no other of the plugins do. All of my website endpoints are currently blank and my grafana dashboards are just showing errors because of morg being down.

I'll reiterate that morg is largely unfinished code with plenty of ideas that were never implemented. With no activity in the last 2 years and having a pull request open for over 8 months. How long will i need to wait before someone can take over the plugin? Or should i just move away from using the morg endpoints completely and hope i can find something else that does the same as morg does?

YvesW commented:
There is no strict amount of time (often 1 month is used though), but we'd like people to be added as contributor to a repo instead of transferring ownership, so you should:
- Create a PR with this in their repo and ask to be added as contributor to the repo.
- Contact the author via any way possible:
  - @ them in the RL discord if you know their discord name and ask to be added as contributor
  - Check their GH for contact info and use that to ask them if you can be added as contributor
  - Check their commits for their email so you can ask them via email if you can be added as contributor
  - Can create an issue as well asking to be added to the repo as contributor

If they haven't responded in 1 month, you can create a new PR here with pictures of them not responding. Also don't forget to @ them in the PR. In that case you'd transfer ownership of the plugin to your own repo, keeping the commit history intact without making any changes. After ownership has been transferred, you can update the plugin to whatever you want.

@MorgApps

raiyni commented:
Morg updated their app, is that satisfactory?

Christley commented:
That's more than satisfactory. I will do pull requests to morg instead and improve that plugin.

Only reason i wrote chrisapi was because i didn't see any activity on morg but now that morg accepts pull requests my plugin is obsolete
